coreUtilities/src/main/java/org/opensearch/migrations/config/Backfill.java

```diff
@@ -0,0 +1,7 @@
+package org.opensearch.migrations.config;
```

Comment:
When I did the instrumentation work, I wanted to make sure that a change needed for one project wouldn't impact in any way any other project that didn't need to consume the change.  At the every-day level, this includes builds and build graphs that are as tight as possible.  It also self-documents the extent that datastructures and patterns are consumed.  If I see Backfill and Cluster, I'm going to presume that every single project that uses parameters will need these.  In fact though, some significant services don't - like the proxy or cluster have no use for many of these and never will.

We have a single repo for this project, but I would pretend that a different person were in the sole maintainer for each project.  If that was the case, would this aggregation be worth the overhead?

Reply:
I agree that any change in coreUtilities requires a rebuild.  And as I understand it its purpose, we put shared functionality between all components in our migration ecosystem.  If the services.yaml is used by all the projects in the solution including the java projects, the place that seems logical to put this shared parsing code would be in the coreUtilities project.

I believe the relationship is worthwhile to be sure that all components can read the same services.yaml definition in the project.  How else would you have a shared definition, where else should it go?